frontend/src/main/web/lib/components/glossary/ActionCell.jsx

```diff
@@ -51,8 +55,34 @@ var ActionCell = React.createClass({
     Actions.resetEntry(this.props.id);
   },
 
-  _onUpdateComment: function (value) {
-    Actions.updateComment(this.props.id, value);
+  _onUpdateComment: function () {
+    this.setState({savingComment: true});
+    Actions.updateComment(this.props.id, this.state.comment);
+  },
+
+  _hasCommentChanged: function() {
+    var initialValue = _.isUndefined(this.state.entry.transTerm.comment) ? '' : this.state.entry.transTerm.comment;
+    var newValue = _.isUndefined(this.state.comment) ? '' : this.state.comment;
+    return initialValue !== newValue;
+  },
+
+  _toggleComment: function () {
+    this.setState({showComment: !this.state.showComment});
+  },
+
+  _onCommentChange: function(event) {
+    this.setState({comment: event.target.value});
+  },
+
+  _onCancelComment: function () {
+    var value = _.isUndefined(this.state.entry.transTerm.comment) ? '' : this.state.entry.transTerm.comment;
+    this.setState({comment: value, showComment: false});
```

Comment:
Question: why is there `state.comment` and `state.entry.transTerm.comment` with the same value? Possible duplication of data?